sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -456,13 +456,13 @@ private void applySessionToken(RxDocumentServiceRequest request) {
                         !Strings.areEqual(requestConsistencyLevel, ConsistencyLevel.EVENTUAL.toString())));
 
         if (!Strings.isNullOrEmpty(request.getHeaders().get(HttpConstants.HttpHeaders.SESSION_TOKEN))) {
-            if (!sessionTokenApplicable || ReplicatedResourceClientUtils.isMasterResource(request.getResourceType())) {
+            if (!sessionTokenApplicable || isMasterOperation(request.getResourceType(), request.getOperationType())) {
```

Comment:
could you add a test to validate when session token is set vs not set?

How to add such test:
You can use ReflectionUtils.getTransportClient() and ReflectionUtils.getTransportClient to register a spy to watch what is sent to the TransportClient. That way you can validate any header in the request including session token.

Reply:
Added tests, a little bit different compared to use ReflectionUtils.TransportClient

Reply:
awesome test for gateway @xinlian12 . later we should add a similar test for direct mode. For that you will need something similar to ReflectionUtils.TransportClient.

As this PR scope is only gateway, doesn't have to happen in this PR. please track similar test for direct for future test improvement.